content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -1,16 +1,168 @@
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findMany({ model, operation, args, data }) {
+        // set `age` and fill with the rest of `where`
+        args.where = { age: { gt: 18 }, ...args.where }
+        const users = await data
+        return data
+      },
+    },
+  },
+})
+```
+
+### Add a custom query to all operations in a specific model
+
+Use `$allOperations` to extend all operations at once instead of a specific one.
+
+For example, the following code applies a custom query to all operations on the `user` model:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      $allOperations({ model, operation, args, data }) {
+        // handle all operations
+        return data
+      },
+    },
+  },
+})
+```
+
+### Add a custom query to all operations for all existing models
+
+Use `$allOperations` to apply your custom query to all operations, instead of to a specific operation name.
+
+To apply a custom query to all operations on all models of your schema:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    $allModels: {
+      $allOperations({ model, operation, args, data }) {
+        // handle all operations
+        return data
+      },
+    },
+  },
+})
+```
+
+## Query extensions versus middlewares
+
+You can use query extensions or [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases) to hook into the query life-cycle and modify an incoming query or its result. Client extensions and middlewares differ in the following ways:
+
+- Middlewares always apply globally to the same client. Client extensions are isolated, unless you deliberately combine them. [Learn more about client extensions](/concepts/components/prisma-client/client-extensions#about-prisma-client-extensions).
+  - For example, in a row-level security (RLS) scenario, you can keep each user in an entirely separate client. With middlewares, all users are active in the same client.
+- During application execution, with extensions you can choose from one or more extended clients, or the standard Prisma Client. With middlewares, you cannot choose which client to use, because there is only one global client.
+- Extensions benefit from end-to-end type safety and inference, but middlewares don't.
+
+You can use Prisma Client extensions in all scenarios where middlewares can be used.
+
+### If you use the <inlinecode>query</inlinecode> extension component and middlewares
+
+If you use the `query` extension component and middlewares in your project, then the following rules and priorities apply:
+
+- In your application code, you must declare all your middlewares on the main Prisma Client instance. You cannot declare them on an extended client.
+- You cannot instantiate middlewares (with `$use`) in extended clients. You can only instantiate middlewares in the original client.
```

Comment:
Is the usage of `$use` forbidden, or does it just appl to all existing Clients (no matter if orignal or extended)?

Reply:
@millsp  Could you comment on this question, too? Thanks.

Reply:
To clarify: Middlewares can only be hooked via a non-extended client. Middlewares don't benefit from extension isolation and will always be executed first in any extended client.

Reply:
I think in this case the "with `$use`" caused the confusion. I don't think we need that detail here, so I'll remove it.